src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -203,6 +207,69 @@ func (r *Cluster) validateKafkaListeners() field.ErrorList {
 	return allErrs
 }
 
+func (r *Cluster) validatePandaproxyListeners() field.ErrorList {
+	var allErrs field.ErrorList
+	var proxyExternal *PandaproxyAPI
+	kafkaExternal := r.ExternalListener()
+	for i, p := range r.Spec.Configuration.PandaproxyAPI {
+		if !p.External.Enabled {
+			continue
+		}
+		if proxyExternal != nil {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"only one pandaproxy api listener can be marked as external"))
+		}
+		proxyExternal = &r.Spec.Configuration.PandaproxyAPI[i]
+		if proxyExternal.Port != 0 {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"external pandaproxy api listener cannot have port specified, it's autogenerated"))
+		}
+		if (kafkaExternal == nil || !kafkaExternal.External.Enabled) && (proxyExternal != nil && proxyExternal.External.Enabled) {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("pandaproxyApi"),
+					r.Spec.Configuration.PandaproxyAPI,
+					"cannot have a pandaproxy external listener without a kafka external listener"))
+		}
+		if kafkaExternal == nil && proxyExternal.External.Subdomain != "" {
```

Comment:
is this correct? So there's no kafka external connectivity but the error message is saying that the subdomains don't match

Reply:
The proxy specifies a subdomain while Kafka is empty, so the subdomains don't match.

I modified the text so it's more direct.